lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -265,6 +268,47 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
     end
   end
 
+  @spec finality_checkpoins_by_id(state_id()) ::
+          {:ok, finality_info()} | {:error, String.t()} | :not_found | :empty_slot | :invalid_id
+  def finality_checkpoins_by_id(hex_root) when is_binary(hex_root) do
+    # TODO compute is_optimistic_or_invalid() and is_finalized()
+    execution_optimistic = true
+    finalized = false
+
+    case StateStore.get_state_by_state_root(hex_root) do
+      {:ok, state} ->
+        %{
+          previous_justified_checkpoint: previous_justified_checkpoint,
+          current_justified_checkpoint: current_justified_checkpoint,
+          finalized_checkpoint: finalized_checkpoint
+        } = state
+
+        {:ok,
+         {previous_justified_checkpoint, current_justified_checkpoint, finalized_checkpoint,
+          execution_optimistic, finalized}}
+
+      _ ->
+        :not_found
+    end
+  end
+
+  def finality_checkpoins_by_id(id) do
+    with {:ok, {block_root, optimistic, finalized}} <- block_root_by_id(id),
+         {:ok, %Types.SignedBeaconBlock{message: %{state_root: state_root}} = _block} <-
+           BlockStore.get_block(block_root),
+         {:ok, state} <- StateStore.get_state_by_state_root(state_root) do
+      %{
+        previous_justified_checkpoint: previous_justified_checkpoint,
+        current_justified_checkpoint: current_justified_checkpoint,
+        finalized_checkpoint: finalized_checkpoint
+      } = state
+
+      {:ok,
+       {previous_justified_checkpoint, current_justified_checkpoint, finalized_checkpoint,
+        optimistic, finalized}}
+    end
+  end
```

Comment:
We shouldn't have a second version here. Let's refactor a bit. Let's have the functions:

- `state_by_state_id(id)`
- `state_root_by_state_id(id)`
- `block_by_block_id(id)`
- `block_root_by_block_id(id)`

And then here we can only have one clause and call

```elixir
with {:ok, state} <- state_by_state_id(id) do
  #...
end
```

and that should work either with the id being a root or whatever else.